docs/elasticsearch/cluster.md

```diff
@@ -198,6 +189,135 @@ x.x.x.x           23          38   0    0.12    0.07     0.06 md        -      o
+
+You also have the option to require that primary and replica shards are never allocated to the same zone. This is called forced awareness.
+
+To configure forced awareness, specify all the possible values for your zone attributes:
+
+```json
+PUT _cluster/settings
+{
+  "persistent": {
+    "cluster.routing.allocation.awareness.attributes": "zone",
+    "cluster.routing.allocation.awareness.force.zone.values":["1", "2"]
+  }
+}
+```
+Now, if one of the data nodes fail, forced awareness will not allocate the replicas to a node in the same zone. Instead, it will keep the cluster in a yellow state and only allocate replicas when nodes in another zone come back online.
+
+You generally want to use allocation awareness when the hardware in each of your zones is less than 50% utilized so that it has the storage capacity to allocate replicas in the same zone.
+
+If you’re in a situation where you do not have the capacity in any one zone to contain the entire cluster with all its primary and replica shards, then you should use forced awareness. So in the event of a failure, you don't over-allocate your last remaining zone and essentially lock up your cluster because it's storage is maxed out.
+
+## Step 7: Set up a hot/warm architecture
+
+You can design a hot/warm architecture where you first index your data to hot nodes---fast and expensive---and after a certain period of time move them to warm nodes---slow and cheap.
+
+This architecture can help you save a lot of money in storage costs because you can increase data retention by increasing the number of warm nodes and add hot nodes based on the total volume of data coming in.
+
+To configure a hot/warm storage architecture, add `temp` attributes to `odfe-d1` and `odfe-d2`, respectively:
+
+```yml
+node.attr.temp: hot
+```
+```yml
+node.attr.temp: warm
+```
+
+You can set the attribute name and value to whatever you want as long as it’s consistent for all your hot and warm nodes.
+
+To add an index `newindex` to the hot node:
+
+```json
+PUT newindex
+{
+  "settings": {
+    "index.routing.allocation.require.temp": "hot"
+  }
+}
+```
+
+If you take a look at the shard allocation for `newindex`:
+
+```json
+GET _cat/shards/newindex?v
+index     shard prirep state      docs store ip         node
+new_index 2     p      STARTED       0  230b 10.0.0.225 odfe-d1
+new_index 2     r      UNASSIGNED
+new_index 3     p      STARTED       0  230b 10.0.0.225 odfe-d1
+new_index 3     r      UNASSIGNED
+new_index 4     p      STARTED       0  230b 10.0.0.225 odfe-d1
+new_index 4     r      UNASSIGNED
+new_index 1     p      STARTED       0  230b 10.0.0.225 odfe-d1
+new_index 1     r      UNASSIGNED
+new_index 0     p      STARTED       0  230b 10.0.0.225 odfe-d1
+new_index 0     r      UNASSIGNED
+```
+
+All primary shards are allocated to `odfe-d1`, which is our hot node. All the replica shards are unassigned because we’re forcing this index to allocate only to hot nodes.
```

Comment:
In this example, all primary shards...

All replica shards are unassigned, because we're forcing this index...